source/opt/private_to_local_pass.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "private_to_local_pass.h"
+
+#include "ir_context.h"
+
+namespace {
+const uint32_t kVariableStorageClassInIdx = 0;
+const int kSpvTypePointerTypeIdInIdx = 1;
+}  // namespace
+
+namespace spvtools {
+namespace opt {
+
+Pass::Status PrivateToLocalPass::Process(ir::IRContext* c) {
+  InitializeProcessing(c);
+  bool modified = false;
+
```

Comment:
The semantics on the lifetime of a private scope variable outside of shaders models are unclear. You should predicate this pass on the shader capabality.

Reply:
From the spec, I think that private variable only exist in shader modules.  I'll add the check, so we don't do pointless work.